agent-inject/agent/annotations.go

```diff
@@ -45,6 +45,15 @@ const (
 	// If not provided, a default generic template is used.
 	AnnotationAgentInjectTemplate = "vault.hashicorp.com/agent-inject-template"
 
+	// AnnotationAgentInjectTemplateSource is the optional key annotation that configures Vault
+	// Agent what template on disk to use for rendering the secrets.  The name
+	// of the template is any unique string after "vault.hashicorp.com/agent-inject-template-source-",
+	// such as "vault.hashicorp.com/agent-inject-template-source-foobar".  This should map
+	// to the same unique value provided in "vault.hashicorp.com/agent-inject-secret-".
+	// The value is the filename and path of the template used by the agent to render the secrets.
+	// If not provided, the template content key annotation is used.
+	AnnotationAgentInjectTemplateSource = "vault.hashicorp.com/agent-inject-template-source"
```

Comment:
Thoughts on making this `vault.hashicorp.com/agent-inject-template-file` instead?

Reply:
Indeed this annotation is clearer. I have updated the code.